### ISSUE TITLE
 Column `release_pos_z` can't be converted from character to numeric

### ISSUE THREAD

**Issue by jestarr** [BillPetti/baseballr#91]
I'm getting the following error when I try to scrape 2019 statcast data:

`r: Column `release_pos_z` can't be converted from character to numeric`

**Comment by BillPetti** [comment]
Works for me. Can you provide a reproduceable example? To me, that looks like an error that can happen when looping and binding different scrapes.

```
> scrape_statcast_savant("2019-04-21", "2019-04-21") %>% head()
2019-04-21 is not a date. Attempting to coerce...
No playerid specified. Collecting data for all batters/pitchers.
https://baseballsavant.mlb.com/statcast_search/csv?all=true&hfPT=&hfAB=&hfBBT=&hfPR=&hfZ=&stadium=&hfBBL=&hfNewZones=&hfGT=R%7CPO%7CS%7C&hfC&hfSea=2019%7C&hfSit=&hfOuts=&opponent=&pitcher_throws=&batter_stands=&hfSA=&player_type=batter&hfInfield=&team=&position=&hfOutfield=&hfRO=&home_road=&game_date_gt=2019-04-21&game_date_lt=2019-04-21&hfFlag=&hfPull=&metric_1=&hfInn=&min_pitches=0&min_results=0&group_by=name&sort_col=pitches&player_event_sort=h_launch_speed&sort_order=desc&min_abs=0&type=details
These data are from BaseballSavant and are property of MLB Advanced Media, L.P. All rights reserved.
Grabbing data, this may take a minute...
URL read and payload acquired successfully.
# A tibble: 6 x 90
  pitch_type game_date  release_speed release_pos_x release_pos_z player_name batter pitcher
  <chr>      <date>             <dbl>         <dbl>         <dbl> <chr>        <dbl>   <dbl>
1 SL         2019-04-21          84.6         -1.20          5.80 Brandon Lo… 664040  518489
2 CH         2019-04-21          87.1         -1.58          5.51 Brandon Lo… 664040  518489
3 SL         2019-04-21          85.1         -1.35          5.83 Brandon Lo… 664040  518489
4 SL         2019-04-21          83.7         -1.38          5.67 Joey Wendle 621563  518489
5 SL         2019-04-21          86.7         -1.35          5.80 Joey Wendle 621563  518489
6 KC         2019-04-21          79.7         -2.46          5.45 Austin Rom… 519222  453178
# … with 82 more variables: events <chr>, description <chr>, spin_dir <lgl>,
#   spin_rate_deprecated <lgl>, break_angle_deprecated <lgl>, break_length_deprecated <lgl>,
#   zone <dbl>, des <chr>, game_type <chr>, stand <chr>, p_throws <chr>, home_team <chr>,
#   away_team <chr>, type <chr>, hit_location <dbl>, bb_type <chr>, balls <dbl>,
#   strikes <dbl>, game_year <dbl>, pfx_x <dbl>, pfx_z <dbl>, plate_x <dbl>, plate_z <dbl>,
#   on_3b <dbl>, on_2b <dbl>, on_1b <dbl>, outs_when_up <dbl>, inning <dbl>,
#   inning_topbot <chr>, hc_x <dbl>, hc_y <dbl>, tfs_deprecated <lgl>,
#   tfs_zulu_deprecated <lgl>, fielder_2 <dbl>, umpire <lgl>, sv_id <chr>, vx0 <dbl>,
#   vy0 <dbl>, vz0 <dbl>, ax <dbl>, ay <dbl>, az <dbl>, sz_top <dbl>, sz_bot <dbl>,
#   hit_distance_sc <dbl>, launch_speed <dbl>, launch_angle <dbl>, effective_speed <dbl>,
#   release_spin_rate <dbl>, release_extension <dbl>, game_pk <dbl>, pitcher_1 <dbl>,
#   fielder_2_1 <dbl>, fielder_3 <dbl>, fielder_4 <dbl>, fielder_5 <dbl>, fielder_6 <dbl>,
#   fielder_7 <dbl>, fielder_8 <dbl>, fielder_9 <dbl>, release_pos_y <dbl>,
#   estimated_ba_using_speedangle <dbl>, estimated_woba_using_speedangle <dbl>,
#   woba_value <dbl>, woba_denom <dbl>, babip_value <dbl>, iso_value <dbl>,
#   launch_speed_angle <dbl>, at_bat_number <dbl>, pitch_number <dbl>, pitch_name <chr>,
#   home_score <dbl>, away_score <dbl>, bat_score <dbl>, fld_score <dbl>,
#   post_away_score <dbl>, post_home_score <dbl>, post_bat_score <dbl>,
#   post_fld_score <dbl>, if_fielding_alignment <chr>, of_fielding_alignment <chr>,
#   barrel <dbl>
```

**Comment by beanumber** [comment]
The problem is that if one of the date ranges returns no results, it errors instead of returning an empty data frame. 

``` r
baseballr::scrape_statcast_savant_batter_all("2019-04-01", "2019-04-01")
#> 2019-04-01 is not a date. Attempting to coerce...
#> No playerid specified. Collecting data for all batters/pitchers.
#> https://baseballsavant.mlb.com/statcast_search/csv?all=true&hfPT=&hfAB=&hfBBT=&hfPR=&hfZ=&stadium=&hfBBL=&hfNewZones=&hfGT=R%7CPO%7CS%7C&hfC&hfSea=2019%7C&hfSit=&hfOuts=&opponent=&pitcher_throws=&batter_stands=&hfSA=&player_type=batter&hfInfield=&team=&position=&hfOutfield=&hfRO=&home_road=&game_date_gt=2019-04-01&game_date_lt=2019-04-01&hfFlag=&hfPull=&metric_1=&hfInn=&min_pitches=0&min_results=0&group_by=name&sort_col=pitches&player_event_sort=h_launch_speed&sort_order=desc&min_abs=0&type=details
#> These data are from BaseballSevant and are property of MLB Advanced Media, L.P. All rights reserved.
#> Grabbing data, this may take a minute...
#> URL read and payload acquired successfully.
#> # A tibble: 3,844 x 90
#>    pitch_type game_date  release_speed release_pos_x release_pos_z
#>    <chr>      <date>             <dbl>         <dbl>         <dbl>
#>  1 FT         2019-04-01          94.1         -2.79          5.64
#>  2 FF         2019-04-01          94.8         -2.65          5.87
#>  3 CU         2019-04-01          75.7         -2.36          6.16
#>  4 SL         2019-04-01          87.3         -2.56          5.94
#>  5 FT         2019-04-01          94.5         -2.78          5.72
#>  6 FT         2019-04-01          95.1         -2.76          5.90
#>  7 CH         2019-04-01          83.6         -2.51          6.01
#>  8 FT         2019-04-01          94.5         -2.75          5.95
#>  9 CU         2019-04-01          76.9         -2.42          6.11
#> 10 FT         2019-04-01          94.9         -2.76          5.90
#> # … with 3,834 more rows, and 85 more variables: player_name <chr>,
#> #   batter <dbl>, pitcher <dbl>, events <chr>, description <chr>,
#> #   spin_dir <lgl>, spin_rate_deprecated <lgl>,
#> #   break_angle_deprecated <lgl>, break_length_deprecated <lgl>,
#> #   zone <dbl>, des <chr>, game_type <chr>, stand <chr>, p_throws <chr>,
#> #   home_team <chr>, away_team <chr>, type <chr>, hit_location <dbl>,
#> #   bb_type <chr>, balls <dbl>, strikes <dbl>, game_year <dbl>,
#> #   pfx_x <dbl>, pfx_z <dbl>, plate_x <dbl>, plate_z <dbl>, on_3b <dbl>,
#> #   on_2b <dbl>, on_1b <dbl>, outs_when_up <dbl>, inning <dbl>,
#> #   inning_topbot <chr>, hc_x <dbl>, hc_y <dbl>, tfs_deprecated <lgl>,
#> #   tfs_zulu_deprecated <lgl>, fielder_2 <dbl>, umpire <lgl>, sv_id <chr>,
#> #   vx0 <dbl>, vy0 <dbl>, vz0 <dbl>, ax <dbl>, ay <dbl>, az <dbl>,
#> #   sz_top <dbl>, sz_bot <dbl>, hit_distance_sc <dbl>, launch_speed <dbl>,
#> #   launch_angle <dbl>, effective_speed <dbl>, release_spin_rate <dbl>,
#> #   release_extension <dbl>, game_pk <dbl>, pitcher_1 <dbl>,
#> #   fielder_2_1 <dbl>, fielder_3 <dbl>, fielder_4 <dbl>, fielder_5 <dbl>,
#> #   fielder_6 <dbl>, fielder_7 <dbl>, fielder_8 <dbl>, fielder_9 <dbl>,
#> #   release_pos_y <dbl>, estimated_ba_using_speedangle <dbl>,
#> #   estimated_woba_using_speedangle <dbl>, woba_value <dbl>,
#> #   woba_denom <dbl>, babip_value <dbl>, iso_value <dbl>,
#> #   launch_speed_angle <dbl>, at_bat_number <dbl>, pitch_number <dbl>,
#> #   pitch_name <chr>, home_score <dbl>, away_score <dbl>, bat_score <dbl>,
#> #   fld_score <dbl>, post_away_score <dbl>, post_home_score <dbl>,
#> #   post_bat_score <dbl>, post_fld_score <dbl>,
#> #   if_fielding_alignment <chr>, of_fielding_alignment <chr>, barrel <dbl>
baseballr::scrape_statcast_savant_batter_all("2019-03-01", "2019-03-01")
#> 2019-03-01 is not a date. Attempting to coerce...
#> No playerid specified. Collecting data for all batters/pitchers.
#> https://baseballsavant.mlb.com/statcast_search/csv?all=true&hfPT=&hfAB=&hfBBT=&hfPR=&hfZ=&stadium=&hfBBL=&hfNewZones=&hfGT=R%7CPO%7CS%7C&hfC&hfSea=2019%7C&hfSit=&hfOuts=&opponent=&pitcher_throws=&batter_stands=&hfSA=&player_type=batter&hfInfield=&team=&position=&hfOutfield=&hfRO=&home_road=&game_date_gt=2019-03-01&game_date_lt=2019-03-01&hfFlag=&hfPull=&metric_1=&hfInn=&min_pitches=0&min_results=0&group_by=name&sort_col=pitches&player_event_sort=h_launch_speed&sort_order=desc&min_abs=0&type=details
#> These data are from BaseballSevant and are property of MLB Advanced Media, L.P. All rights reserved.
#> Grabbing data, this may take a minute...
#> URL read and payload acquired successfully.
#> Error: Evaluation error: non-numeric argument to binary operator.
```

<sup>Created on 2019-04-23 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

It should be easy enough to fix, but don't have the time right this second.

**Comment by BillPetti** [comment]
Okay, I see the issue now. @jestarr is probably using one of the older functions (that I really should formally deprecate); i.e. `scrape_statcast_savant_batter` or `scrape_statcast_savant_pitcher`. 

The updated/inclusive `scrape_statcast_savant` will produce an empty tibble if no data is available for a given date:

```
> baseballr::scrape_statcast_savant("2019-03-01", "2019-03-01", player_type = "batter")
2019-03-01 is not a date. Attempting to coerce...
No playerid specified. Collecting data for all batters/pitchers.
https://baseballsavant.mlb.com/statcast_search/csv?all=true&hfPT=&hfAB=&hfBBT=&hfPR=&hfZ=&stadium=&hfBBL=&hfNewZones=&hfGT=R%7CPO%7CS%7C&hfC&hfSea=2019%7C&hfSit=&hfOuts=&opponent=&pitcher_throws=&batter_stands=&hfSA=&player_type=batter&hfInfield=&team=&position=&hfOutfield=&hfRO=&home_road=&game_date_gt=2019-03-01&game_date_lt=2019-03-01&hfFlag=&hfPull=&metric_1=&hfInn=&min_pitches=0&min_results=0&group_by=name&sort_col=pitches&player_event_sort=h_launch_speed&sort_order=desc&min_abs=0&type=details
These data are from BaseballSavant and are property of MLB Advanced Media, L.P. All rights reserved.
Grabbing data, this may take a minute...
URL read and payload acquired successfully.
# A tibble: 0 x 90
# … with 90 variables: pitch_type <chr>, game_date <date>, release_speed <dbl>,
#   release_pos_x <dbl>, release_pos_z <chr>, player_name <chr>, batter <chr>,
#   pitcher <chr>, events <chr>, description <chr>, spin_dir <chr>,
#   spin_rate_deprecated <chr>, break_angle_deprecated <chr>,
#   break_length_deprecated <chr>, zone <dbl>, des <chr>, game_type <chr>,
#   stand <chr>, p_throws <chr>, home_team <chr>, away_team <chr>, type <chr>,
#   hit_location <chr>, bb_type <chr>, balls <chr>, strikes <chr>, game_year <chr>,
#   pfx_x <dbl>, pfx_z <dbl>, plate_x <dbl>, plate_z <dbl>, on_3b <dbl>,
#   on_2b <dbl>, on_1b <dbl>, outs_when_up <chr>, inning <chr>,
#   inning_topbot <chr>, hc_x <dbl>, hc_y <dbl>, tfs_deprecated <chr>,
#   tfs_zulu_deprecated <chr>, fielder_2 <chr>, umpire <chr>, sv_id <chr>,
#   vx0 <dbl>, vy0 <dbl>, vz0 <dbl>, ax <dbl>, ay <dbl>, az <dbl>, sz_top <dbl>,
#   sz_bot <dbl>, hit_distance_sc <dbl>, launch_speed <dbl>, launch_angle <dbl>,
#   effective_speed <dbl>, release_spin_rate <dbl>, release_extension <dbl>,
#   game_pk <dbl>, pitcher_1 <chr>, fielder_2_1 <chr>, fielder_3 <chr>,
#   fielder_4 <chr>, fielder_5 <chr>, fielder_6 <chr>, fielder_7 <chr>,
#   fielder_8 <chr>, fielder_9 <chr>, release_pos_y <chr>,
#   estimated_ba_using_speedangle <chr>, estimated_woba_using_speedangle <chr>,
#   woba_value <chr>, woba_denom <chr>, babip_value <chr>, iso_value <chr>,
#   launch_speed_angle <chr>, at_bat_number <chr>, pitch_number <chr>,
#   pitch_name <chr>, home_score <dbl>, away_score <dbl>, bat_score <dbl>,
#   fld_score <dbl>, post_away_score <dbl>, post_home_score <dbl>,
#   post_bat_score <dbl>, post_fld_score <dbl>, if_fielding_alignment <chr>,
#   of_fielding_alignment <chr>, barrel <lgl>
```

**Comment by lfbv** [comment]
I also have encountered this when merging datasets scraped with the new function (scrape_statcast_savant). The empty tibble that is produced lists release_pos_z as character, as you can see in your output example above.

**Comment by BillPetti** [comment]
This has been fixed by commit [aeb17058f167ba948b367336f5cf2ba80ecbc716](https://github.com/BillPetti/baseballr/commit/aeb17058f167ba948b367336f5cf2ba80ecbc716)
